plugins/modules/deb_publication.py

```diff
@@ -26,6 +26,22 @@
       - Version number to be published
     type: int
     required: false
+  simple:
+    description:
+      - Whether to use simple mode. If none of simple, structured or verbatim
+        are specified, then a simple publication will be created
+    type: bool
+    required: false
```

Comment:
If we use "default: False" instead of "required: False", wouldn't it make the code much cleaner?
Maybe a bit confusing with implicit simple however...

Reply:
I don't think it accurately represents the default though. The default is True, unless you specify any mode options.

Reply:
That's what i mean by 'confusing' and also what kind of pushes me to the choice option, where it's easy to say `default: "simple"`, and users reading the docs will immediately understand.

Reply:
Are there likely to be further modes added? It could lead to a combinatorial explosion of options

Reply:
Another option is to make a separate module for verbatim pubs.

Reply:
I currently don't see a need for more modes.

Reply:
So we currently have three options:
- One module, with three separate boolean options (simple, structured, verbatim) that have relatively complex relations. (Current draft implementation).
- Two modules (separate module for verbatim pubs). This would mirror the structure of the pulp_deb API exactly, however it has proven to be difficult to understand and document.
- One module with one choice option that would need to include all useful combinations as a distinct choice.

If it is between option one and two, I would go for the separate verbatim module, and the one to one relation with the API endpoints. However, I actually think the third option is best. I am not to worried about "combinatorial explosion". New publication modes in the future are not unthinkable, but are expected to be rare. If we do add new publication modes in the future, I would probably add another separate endpoint like verbatim, rather than adding even more complexity to the APT publisher. That way we would get exactly one additional mode, and not a "combinatorial explosion". Note that the APT publisher is probably misnamed, since all the pulp_deb publishers are APT publishers, duh! What I am currently calling the APT publisher, is really the simple and/or structured publisher, but that is a mouthful.

Taking all this together my suggestion is something like:
```
mode=dict(default="structured", choices=["simple", "structured", "simple and structured", "verbatim"]),
```
This makes it very clear what the choices are.

Note that I have suggested "structured" as the default. This is a separate issue, but I feel like structured is perhaps the better default, since it recreates the structure of the upstream repo that was synced, which is probably what a naive user would expect from the default behaviour. (https://en.wikipedia.org/wiki/Principle_of_least_astonishment)

Reply:
Thanks for the response. I'm happy to go with option 3. I agree that structured is a better default, although it would break backwards compatibility.

Reply:
About structured. It currently does not publish uploaded packages in the way you'd expect, so what is the bigger surprise?

Reply:
Note sure if whitespace in a choice field is desirable, if not, just make it "simple_and_structured" or something...

Reply:
Ok, maybe leave the default as simple then. We can always leave that as a discussion for a later time.

Reply:
Or make the default "simple and structured", then everyone is equally unhappy! (No seriously let's not mess with this as part of this change).